src/django/api/facility_actions/processing_facility_list.py

```diff
@@ -0,0 +1,229 @@
+    '''
+    Class to process a facility list.
+    '''
+
+    def __init__(self, processing_input: Dict[str, Any]) -> None:
+        self.__uploaded_file: Union[
+            InMemoryUploadedFile, TemporaryUploadedFile, Dict
+        ] = processing_input['uploaded_file']
+        self.__processed_data: ListDTO = processing_input['processed_data']
+        self.__contributor: Contributor = processing_input['contributor']
+        self.__parsing_started: str = processing_input['parsing_started']
+        self.__serializer_method: MethodType = processing_input[
+            'serializer_method'
+        ]
+        self.__name: str = processing_input['name']
+        self.__description: str = processing_input['description']
+        self.__replaces: Union[None, FacilityList] = processing_input[
+            'replaces'
+        ]
+        self.__request: Request = processing_input['request']
+
+    def process_facility(self) -> Response:
+        # handle processing errors
+        if self.__processed_data.errors:
+            log.error(
+                '[List Upload] CC Validation Errors: '
+                f'{self.__processed_data.errors}'
+            )
+            error_messages = [
+                str(error['message']) for error in self.__processed_data.errors
+            ]
+            raise ValidationError(error_messages)
+
+        rows = self.__processed_data.rows
+        header_row_keys = rows[0].raw_json.keys()
+        header_str = ','.join(header_row_keys)
+
+        new_list = self.__create_list(header_str)
+        log.info(f'[List Upload] FacilityList created. Id {new_list.id}!')
+
+        self._create_nonstandard_fields(header_row_keys, self.__contributor)
+
+        source = self._create_source(new_list)
+        log.info(f'[List Upload] Source created. Id {source.id}!')
+
+        is_geocoded: bool = False
+        parsed_items: Set[str] = set()
+
+        for idx, row in enumerate(rows):
+            # Created a partially filled FacilityListItem to save valid data
+            # and to provide an item for saving any errors that may exist
+            # below.
+            item = self._create_facility_list_item(
+                source, row, idx, header_str
+            )
+            log.info(f'[List Upload] FacilityListItem created. Id {item.id}!')
+
+            self.__handle_row_errors(item, row)
+
+            if item.status != FacilityListItem.ERROR_PARSING:
+                item.sector = row.sector
+                item.country_code = row.country_code
+                item.name = row.name
+                item.clean_name = row.clean_name
+                item.address = row.address
+                item.clean_address = row.clean_address
+                try:
+                    self.__process_valid_item(item, row)
+                except Exception as e:
+                    self.__handle_processing_exception(item, e)
+
+            self.__finalize_item_processing(item, is_geocoded, parsed_items)
```

Comment:
Most of the parts of this method have been moved to separate methods. Let's create a special private method for this block of code 83 - 109 line.

Reply:
I'd offer to make this as separate task. Let's create a ticket and resolve this conversation?
@vladsha-dev are you okay?